website/docs/guides/catalog_subscription_and_sharing.html.markdown

```diff
@@ -0,0 +1,342 @@
+  catalog_id = vcd_subscribed_catalog.subscriber.id
+  name       = vcd_subscribed_catalog.subscriber.vapp_template_list[count.index]
+}
+
+data "vcd_catalog_media" "subscribed_media" {
+  count   = vcd_subscribed_catalog.subscriber.number_of_media
+  org     = "other-org"
+  catalog = vcd_subscribed_catalog.subscriber.name
+  name    = vcd_subscribed_catalog.subscriber.media_item_list[count.index]
+}
+
+output "templates" {
+  value = data.vcd_catalog_vapp_template.subscribed_templates
+}
+
+output "media" {
+  value = data.vcd_catalog_media.subscribed_media
+}
+```
+
+Note: If the synchronisation has not happened yet, the data source definition may fail, as the subscribed catalog has not
+received the items. If that happens, a refresh of the resource will solve the issue.
+
+### The role of catalog items
+
+While we work directly with vApp templates and media items, behind the scenes both these entities are catalog items.
+Most of the time, this distinction is unimportant. However, it's worth noting a few concepts:
+
+* The synchronisation happens with catalog items, which in turn contain either a vApp template or a media item;
+* When a catalog has been synchronised, but not the individual items, the vApp templates and media items contained
+  in the catalog items are **temporary**. As soon as each catalog item gets synchronised, the underlying entity
+  (vApp template or media) is deleted and the real one is created through the subscription. You can observe this
+  behavior in the web interface. As soon as you ask to synchronise one item, you will see a task for its deletion,
+  immediately followed by a creation/synchronisation task.
+
+The above points are important in one particular case: if you synchronise the catalog (`sync_catalog` property) but not
+the vApp templates, and try creating a data source using the ID of a temporary entity, you will get an error.
+
+
+## Troubleshooting
+
+### Stuck on deletion: side effects of "sync\_on\_refresh"
+
+We said before that the most efficient way of keeping the subscribed catalog synchronised is to use `sync_on_refresh`,
+allowing a synchronisation every time the catalog is read, which means `terraform plan`, `terraform refresh`, and
+`terraform apply`, even if no properties were altered since the last apply.
+This property is a workaround to a well known Terraform limitation, which does not allow to update a resource unless
+its schema has been changed.
+The workaround has the side effect that a new synchronisation will start on almost every terraform command, including,
+oddly enough, `terraform destroy`. The first operation before apply and destroy is a refresh, whether we ask for it or
+not. In our case, this situation will trigger a synchronisation right when we don't need it, since we have decided
+to remove the subscribed catalog.
+
+To avoid this side effect, we need to remove `sync_on_refresh` from both the configuration file **and from the state file**
+(`terraform.tfstate`).
```

Comment:
To make it totally clear (and for the users who are just eye-scanning the docs):

```suggestion
To avoid this side effect, we need to remove `sync_on_refresh` from both the configuration file **and from the state file**
(`terraform.tfstate`) before calling `terraform destroy`.
```

p.s. This whole document is a super useful read, thanks for such a solid guide!

Reply:
Updated